src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class MainIntegrationTest {
+    private static final String RESOURCE_DIR = "src/test/resources/";
+    private static final String TEST_DIR = "test/";
+    private static final String MANUAL_DIR = "manual/";
+    private static final String TEST_CLASS_PACKAGE_NAME = "test.";
+    private static final String MANUAL_CLASS_PACKAGE_NAME = "manual.";
+    private static final String PACKAGE_TEST_NAME = "test";
+    private static final IOFileFilter IO_FILE_FILTER = new IOFileFilter() {
+        @Override
+        public boolean accept(File file) {
+            return false;
+        }
+
+        @Override
+        public boolean accept(File dir, String name) {
+            return false;
+        }
+    };
+
+    private boolean checkLocatorUniqueness;
+    private boolean forceGenerateFiles;
+    private String inputJavaFileTest;
+    private String inputJavaFileManual;
+    private String generatedClassFullName;
+    private String manualClassFullName;
+    private String jsonSourcePath;
+    private String url;
+    private Class testClass;
+    private Class manualClass;
+
+    private static List<File> filesList = new ArrayList<>();
+    private List<TestClassDTO> classesList;
+
+    @SuppressWarnings("Duplicates")
+    private PageObjectsGenerator initPog(String jsonPath, String url,
+                                         boolean checkLocatorUniqueness,
+                                         boolean forceGenerateFiles) throws IOException {
+        List<String> urls = new ArrayList<>();
+        urls.add(url);
+        SupportedTypesContainer bc = new SupportedTypesContainer();
+        JsonRuleMapper parser = new JsonRuleMapper(new File(jsonPath), new ObjectMapper());
+        XpathToCssTransformation xpathToCssTransformation = new XpathToCssTransformation();
+        JavaPoetAdapter javaPoetAdapter = new JavaPoetAdapter(bc, xpathToCssTransformation);
+
+        ValidatorsStarter validatorsStarter = new ValidatorsStarter(bc);
+        validatorsStarter.setCheckLocatorsUniqueness(checkLocatorUniqueness);
+
+        PageObjectsGenerator pog = new PageObjectsGenerator(parser, validatorsStarter,
+                javaPoetAdapter, RESOURCE_DIR, urls, PACKAGE_TEST_NAME);
+        pog.setForceGenerateFile(forceGenerateFiles);
+        return pog;
+    }
+
+    public MainIntegrationTest(List<TestClassDTO> list) throws MalformedURLException {
+        this.classesList = list;
+    }
+
+    @Parameters
+    public static Iterable<Object[]> data() throws IOException, ClassNotFoundException {
+
+        List<File> caseDirs = getMainDirsNames();
+        List<Object[]> testParameters = new ArrayList<>();
+
+        for (File caseDir : caseDirs) {
+            List<TestClassDTO> objectList = new ArrayList<>();
+            Properties caseProperties = new Properties();
+
+            List<File> insideDirs = (List<File>) FileUtils.listFilesAndDirs(
```

Comment:
It is bad practice to downcast result of external library, especially when library version is not fixed, but set to LATEST.

Reply:
fixed.